src/System Application/App/Azure Key Vault/src/AzureKeyVaultImpl.Codeunit.al

```diff
@@ -30,18 +30,25 @@ codeunit 2202 "Azure Key Vault Impl."
         IsKeyVaultClientInitialized: Boolean;
         AzureKeyVaultTxt: Label 'Azure Key Vault', Locked = true;
         CertificateInfoTxt: Label 'Successfully constructed certificate from secret %1. Certificate thumbprint %2', Locked = true;
+        MissingSecretErr: Label '%1 is either missing or empty', Comment = '%1 = Secret Name.';
 
     [NonDebuggable]
     procedure GetAzureKeyVaultSecret(SecretName: Text; var Secret: Text)
     begin
         // Gets the secret as a Text from the key vault, given a SecretName.
         Secret := GetSecretFromClient(SecretName);
+
+        if Secret = '' then
```

Comment:
perhaps: "if Secret.Trim() = '' then"?
I see we do that in many places, I assume there's a reason for this. Also if we do this change, we can fix all those calls to not make this check anymore :)

Reply:
This is actually added because the platform fetch secrets in a try/catch, so the old way will never fail unless the secret was not in the allowed list. Since we are getting rid of this pattern, the easist way to have a reasonable pattern is to throw an exception if secret is empty/not found. Either way, your feature should break if that secret does not exist or empty. 

Trimming is not a bad idea, let me add it.